### ISSUE TITLE
Camera unavailable and shaky video calls even though live preview works correctly - OnePlus 12

### ISSUE THREAD

**Issue by pikawika** [signalapp/Signal-Android#13440]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
On my OnePlus 12, Android 14, Latest signal app from play store (6.46.7), I get a "Camera unavailable" toast message when I try to capture a photo from the front camera.
This happens only when I press the "capture" button, the actual live preview feed works fine and without issue.
Once the shutter button is pressed once, the other buttons become unresponsive after the toast message appeared.
The back camera does not have this issue.
As per other's comments, i.e. [this one](https://github.com/signalapp/Signal-Android/issues/11465#issuecomment-1952452419) on comparable issues, I noticed multiple OnePlus 12 users are having this issue.

Additionally, when i video call someone, the front camera works but has a very "shaky" feed when moving the camera.
Almost like the framerate is low, eventhough it seems fine if the phone/camera is not moving.
This shaking issue is not present in the live preview of the signal in-app photo preview.

### Steps to reproduce
For the "Camera unavailable" toast message:

1. Click the camera icon in any chat
2. Press the shutter button
3. After a ~1 second freeze of the frame, the "Camera unavailable" toast message pops up
4. The live feed remains to work correctly, but the camera shutter or switch to back camera button become unresponsive.

For the shaky video:

1. Video call anyone
2. Move the phone/camera
3. A shaky feed will be shown, almost as if the camera has a low frame rate, even though if the camera is not moving the frame rate seems normal

**Actual result:** frame freezes for ~1 second, "Camera unavailable" toast message pops up, feed works again but buttons become unresponsive.
**Expected result:** A picture from the front camera is taken.

### Screenshots

![signal_error](https://github.com/signalapp/Signal-Android/assets/32330307/86ee6f88-9d26-403e-9b4d-2c0fac405916)

### Device info
**Device:** OnePlus 12, 16+512 GB version
**Android version:** 14
**Signal version:** 6.46.7 (also happened on previous versions)

### Link to debug log

[here](https://debuglogs.org/android/6.46.7/5902d0213dff676ccdd10cc100210ff6463b0765728f401ae7b94c3cb614fc69)

**Comment by Ramorous** [comment]
OP12 as well and can confirm and reproduce the issue.

**Comment by nicholas-signal** [comment]
We are unable to reproduce this on any of our devices, and appears to be a OnePlus 12-specific bug. I did overhaul the camera controller in 7.1, so when that is available, I would appreciate those with OnePlus 12s to test it.

**Comment by pikawika** [comment]
> I would appreciate those with OnePlus 12s to test it.

Will definitely update here if something changes. Currently all is still very much the same. For what it's worth, apps such as WhatsApp etc don't have this issue.

**Comment by Ramorous** [comment]
In beta and on 7.0.2, will test and confirm when 7.1 is available

**Comment by Ramorous** [comment]
Still running 7.0.2 and the latest OS update resolved the camera issue for me.

**Comment by pikawika** [comment]
Weird, I have the EU variant of that OS (CPH2581_14.0.0.502) and same signal version and same issues persists for me. Anyways, does seem like it is a OnePlus issue then.

**Comment by Ramorous** [comment]
v7.1.1 issue persists

**Comment by thefinn93** [comment]
I'm seeing a similar issue on a Pixel 4A 5G (bramble) with Signal 7.1.1. Not getting any issues with video calls, but can't take pictures with the rear-facing camera. Same symptoms listed above: live preview works, but pressing the shutter button creates a "camera unavailable" toast and doesn't take a picture.

**Comment by Icexist** [comment]
Pixel 4a 5g (android 14) and Signal 7.1.3 with the same issue as thefinn93. Front camera works fine. Video calls work fine with both cameras. Rear camera shows "camera unavailable" toast.

**Comment by cpu2910** [comment]
Same for my pixel5, selfie camera is working , front camera not. Whatsapp and others without problems.

**Comment by Icexist** [comment]
Interestingly, i've been testing this a bit more and can compare two different sequences of actions here:

## Rear Camera Photo

1. Tap the camera icon in the message input.
2. Tap camera shutter button
3. Preview Freezes
4. "Camera Unavailable" toast
5. Preview Unfreezes
6. Toast goes away
7. Subsequent taps of shutter don't work
8. Pressing and holding for video also doesn't work

## Rear Camera Video

1. Tap camera icon in message input
2. Press and hold shutter to record video
3. Video records successfully!

In the second case, if you exit out of the video sending (back action) and try the camera for a still shot, it shows the error toast.

**Comment by kozality** [comment]
Adding to Icexist's comments. I'm using a Pixel 5, and I can confirm the same behavior. Interestingly, my front camera continues to function. Although, I can no longer switch to the front camera if I have already unsuccessfully tried taking a photo with the rear one.

**Comment by CarlB-dev** [comment]
Pixel 5 Android 14 Signal version 7.1.3 and I too am having the same issue.  The selfie camera works in Signal with no issue, however the rear camera displays but when you tap the button to take a picture I too see the camera unavailable.

**Comment by nicholas-signal** [comment]
Hello all, we just released 7.2.0 to beta. Please give it a try and let me know if this issue is fixed!

**Comment by kozality** [comment]
> Hello all, we just released 7.2.0 to beta. Please give it a try and let me know if this issue is fixed!

Working now with 7.2beta. =D

**Comment by pikawika** [comment]
> Hello all, we just released 7.2.0 to beta. Please give it a try and let me know if this issue is fixed!

Sadly, on my OnePlus 12, the issue persist in 7.2.0. Identical behaviour as described in original problem of this thread. Hoping for a fix!

**Comment by CarlB-dev** [comment]
Pixel 5 Android 14 signal 7.2 beta FIXED

**Comment by Icexist** [comment]
Fixed for me Pixel 4a5g, signal 7.2 beta.

**Comment by CarlB-dev** [comment]
I thought the same thing but discovered that the beta had not downloaded
yet.

**Comment by LunaticoCR** [comment]
I've been having the exact same issues on a Pixel 5 as described above. After reading this issue, I've joined beta and received the update and now it's working fine. Thank you!

**Comment by Zanop** [comment]
OnePlus 12
Signal version 7.3.1
OxygenOS 14.0.0.604(EX01)

Same issue with front camera.

**Comment by BlackSheepbe** [comment]
Still having issue on OnePlus 12 and 7.5.2 app version.

**Comment by pikawika** [comment]
Camera seems to work in 7.6.2 on my OnePlus 12!

**Comment by greyson-signal** [comment]
Yeah, I've played the OnePlus 12 on our CameraX blocklist, meaning it's falling back to the older camera implementation for now. The downside is that it can no longer record video in-app. I'll have to o continue to investigate what's going on.

**Comment by pikawika** [comment]
Any update @greyson-signal?

Camera indeed now works, both for taking pictures AND videos!
However, during video calls, there is still a weird artefact present. I think I can best describe it as a malfunctioning algorithm for smooth camera movement? You move the phone during a call and you can clearly see the video movement lasts longer then the actual phone movement, with the video movement being "smoothed",  the video feed is also always zoomed in during video calls, I'm guessing to allow for this software smoothing? This looks weird at best for large movements and trash for smaller movements. When just holding the phone, the minute movements in your hand trigger a continuous smoothing effect resulting in a shaking/wobbly effect of the camera. This is not present taking video's via the app, where the video is also not cropped and this smoothing algorithm doesn't seem present.

**Comment by pikawika** [comment]
https://youtube.com/shorts/DLdZgJ4o_C4

In this video you can see a screen recording of

1. The in-app recording of a video, working as expected
2. The in-app video calling feed being cropped in with some heavy shaking due to what I think is a malfunctioning smoothing algorithm

Both videos from same distance with same amount of movement, arguably even less movement in the second example.
